packages/@storefront/breadcrumbs/src/refinement-crumbs/index.ts

```diff
@@ -10,10 +10,16 @@ class RefinementCrumbs {
       case StoreSections.PAST_PURCHASES:
         this.state.selectedRefinementsUpdated = Events.PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED;
         this.state.navigationSelector = (field) => this.select(Selectors.pastPurchaseNavigation, field);
+        if (!this.props.selectedNavigation) {
+          this.subscribe(Events.PAST_PURCHASE_NAVIGATIONS_UPDATED, this.updateRefinements);
+        }
         break;
       case StoreSections.SEARCH:
         this.state.selectedRefinementsUpdated = Events.SELECTED_REFINEMENTS_UPDATED;
         this.state.navigationSelector = (field) => this.select(Selectors.navigation, field);
+        if (!this.props.selectedNavigation) {
+          this.subscribe(Events.NAVIGATIONS_UPDATED, this.updateRefinements);
```

Comment:
I still don't think you should introduce a pattern to this code base that we intend to rip out and can't because we introduced it and now have to support it